From 91cfc5dcb347ddab8622060de1a588c7a12fe9f9 Mon Sep 17 00:00:00 2001
From: Charlie Paul <cpaul.windriver@gmail.com>
Date: Tue, 11 Aug 2015 21:01:19 -0700
Subject: [PATCH 12/59] drivers/dma: Updated to support axxia dma

git://git.yoctoproject.org/linux-yocto-4.1 standard/axxia/base
	commit 553fd5bb2cfd00cf047ffa6fbdfb3be931909883

Changes to support the axxia dma.

Signed-off-by: Charlie Paul <cpaul.windriver@gmail.com>
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/dma/lsi-dma32.c   |    2 +-
 include/linux/dmaengine.h |   20 ++++++++++++++++++++
 2 files changed, 21 insertions(+), 1 deletions(-)

diff --git a/drivers/dma/lsi-dma32.c b/drivers/dma/lsi-dma32.c
index 6d1cc1a..7604039 100644
--- a/drivers/dma/lsi-dma32.c
+++ b/drivers/dma/lsi-dma32.c
@@ -783,7 +783,7 @@ static int gpdma_of_probe(struct platform_device *op)
 	dma->device_prep_dma_memcpy = gpdma_prep_memcpy;
 	dma->device_prep_dma_sg = gpdma_prep_sg;
 	dma->device_issue_pending = gpdma_issue_pending;
-	dma->device_control = gpdma_device_control;
+//	dma->device_control = gpdma_device_control;
 	INIT_LIST_HEAD(&dma->channels);
 
 	/* Map device I/O memory
diff --git a/include/linux/dmaengine.h b/include/linux/dmaengine.h
index ad41975..4b218df 100644
--- a/include/linux/dmaengine.h
+++ b/include/linux/dmaengine.h
@@ -185,6 +185,26 @@ enum dma_ctrl_flags {
 };
 
 /**
+ * enum dma_ctrl_cmd - DMA operations that can optionally be exercised
+ * on a running channel.
+ * @DMA_TERMINATE_ALL: terminate all ongoing transfers
+ * @DMA_PAUSE: pause ongoing transfers
+ * @DMA_RESUME: resume paused transfer
+ * @DMA_SLAVE_CONFIG: this command is only implemented by DMA controllers
+ * that need to runtime reconfigure the slave channels (as opposed to passing
+ * configuration data in statically from the platform). An additional
+ * argument of struct dma_slave_config must be passed in with this
+ * command.
+ */
+enum dma_ctrl_cmd {
+	DMA_TERMINATE_ALL,
+	DMA_PAUSE,
+	DMA_RESUME,
+	DMA_SLAVE_CONFIG,
+};
+
+
+/**
  * enum sum_check_bits - bit position of pq_check_flags
  */
 enum sum_check_bits {
-- 
1.7.5.4

